src/go/rpk/pkg/cli/cmd/redpanda/admin/brokers/decommission.go

```diff
@@ -37,15 +38,32 @@ leader handles the request.
 			cl, err := admin.NewClient(fs, cfg)
 			out.MaybeDie(err, "unable to initialize admin client: %v", err)
 
-			b, err := cl.Broker(cmd.Context(), broker)
-			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+			brokers, err := cl.Brokers(cmd.Context())
+			out.MaybeDie(err, "unable to get broker list: %v", err)
+
+			var b admin.Broker
+			var found bool
+			for _, br := range brokers {
+				if br.NodeID == broker {
+					b, found = br, true
+					break
+				}
+			}
+			if !found {
+				out.Die("unable to find broker %v in the cluster", broker)
+			}
+
+			version, err := redpanda.VersionFromString(b.Version)
+			out.MaybeDie(err, "unable to get broker version: %v", err)
 
-			// Old brokers (< v22.1) don't have maintenance mode, so we must
-			// check if b.Maintenance is not nil.
-			if b.Maintenance != nil && b.Maintenance.Draining {
-				out.Die(`Node cannot be decommissioned while it is in maintenance mode.
+			if !version.IsHigherOrEqual(redpanda.Version{Year: 23, Feature: 1, Patch: 1}) {
+				// Old brokers (< v22.1) don't have maintenance mode, so we must
```

Comment:
I think they still have maintenance mode, but v23.1+ doesn't have the problem of decom+maintenance resulting in a permanently stuck broker.

Reply:
I see, comment is explicit about pre-22.1, which don't have this entirely